torch/csrc/jit/register_prim_ops.cpp

```diff
@@ -1003,6 +1003,31 @@ Operation listAppend(const Node* node) {
   };
 }
 
+template <typename TList>
+Operation listPop(const Node* node) {
+  return [](Stack& stack) {
+    TList list;
+    int64_t idx;
+    pop(stack, list, idx);
+
+    auto& elements = list->elements();
+    
+    if (elements.size() == 0) {
+      AT_ERROR("pop from empty list");
+    }
+
+    if (idx == -1) {
+      push(stack, std::move(elements.back()));
+      elements.pop_back();
+    } else {
+      push(stack, std::move(getItem(list, idx)));
+      elements.erase(elements.begin() + idx);
```

Comment:
pop can take in negative indices other than -1, so the idx needs to be normalized before the call to erase (and you won't need a special case for idx == -1).